tests/compare_methods_deleonberne.py

```diff
@@ -7,151 +7,30 @@
-    
-    F = [-dVdx, -dVdy]
-    
-    return F
-
-def pot_energy_deleonberne(x, y, par):
-    """Returns the potential energy function V(x,y)
-    """
-    
-    return par[3]*( 1 - np.exp(-par[4]*x) )**2 + 4*y**2*(y**2 \
-        - 1)*np.exp(-par[5]*par[4]*x) + par[2]
-
-
-def variational_eqns_deleonberne(t,PHI,par):
-    """    
-    Returns the state transition matrix , PHI(t,t0), where Df(t) is the Jacobian of the 
-    Hamiltonian vector field
-    
-    d PHI(t, t0)
-    ------------ =  Df(t) * PHI(t, t0)
-        dt
-    
-    """
-    
-    phi = PHI[0:16]
-    phimatrix  = np.reshape(PHI[0:16],(4,4))
-    x,y,px,py = PHI[16:20]
-    
-    
-    # The first order derivative of the potential energy.
-    dVdx = - 2*par[3]*par[4]*np.exp(-par[4]*x)*(np.exp(-par[4]*x) - 1) \
-        - 4*par[5]*par[4]*y**2*(y**2 - 1)*np.exp(-par[5]*par[4]*x) 
-    dVdy = 8*y*(2*y**2 - 1)*np.exp(-par[5]*par[4]*x)
-
-    # The second order derivative of the potential energy. 
-    d2Vdx2 = - ( 2*par[3]*par[4]**2*( np.exp(-par[4]*x) \
-                - 2.0*np.exp(-2*par[4]*x) ) \
-                - 4*(par[5]*par[4])**2*x**2*(y**2 \
-                - 1)*np.exp(-par[5]*par[4]*x) )
-        
-    d2Vdy2 = 8*(6*y**2 - 1)*np.exp( -par[4]*par[5]*x )
-
-    d2Vdydx = -8*y*par[4]*par[5]*np.exp( -par[4]*par[5]*x )*(2*y**2 - 1)
-        
-    
-    
-    d2Vdxdy = d2Vdydx    
-
-    Df    = np.array([[  0,     0,    par[0],    0],
-              [0,     0,    0,    par[1]],
-              [-d2Vdx2,  -d2Vdydx,   0,    0],
-              [-d2Vdxdy, -d2Vdy2,    0,    0]])
-
-    
-    phidot = np.matmul(Df, phimatrix) # variational equation
-
-    PHIdot        = np.zeros(20)
-    PHIdot[0:16]  = np.reshape(phidot,(1,16)) 
-    PHIdot[16]    = px/par[0]
-    PHIdot[17]    = py/par[1]
-    PHIdot[18]    = -dVdx 
-    PHIdot[19]    = -dVdy
-    
-    return list(PHIdot)
-
-
-def ham2dof_deleonberne(t, x, par):
-    """ 
-    Returns the Hamiltonian vector field (Hamilton's equations of motion) 
-    """
-    
-    xDot = np.zeros(4)
-    
-    dVdx = -2*par[3]*par[4]*np.exp(-par[4]*x[0])*(np.exp(-par[4]*x[0]) - 1) - \
-        4*par[5]*par[4]*x[1]**2*(x[1]**2 - 1)*np.exp(-par[5]*par[4]*x[0])
-    dVdy = 8*x[1]*(2*x[1]**2 - 1)*np.exp(-par[5]*par[4]*x[0])
-        
-    xDot[0] = x[2]/par[0]
-    xDot[1] = x[3]/par[1]
-    xDot[2] = -dVdx 
-    xDot[3] = -dVdy
-    
-    return list(xDot)  
-
-
-
-def half_period_deleonberne(t,x,par):
-    """
-    Returns the turning point where we want to stop the integration                           
-    
-    pxDot = x[0]
-    pyDot = x[1]
-    xDot = x[2]
-    yDot = x[3]
-    """
-    
-    return x[2]
-        
-half_period_deleonberne.terminal = True 
-# The zero can be approached from either direction
-half_period_deleonberne.direction=0#0: all directions of crossing
-
-#% End problem specific functions
-
+import os
+path_to_data = os.path.join(os.path.dirname(__file__), '../data/')

```

Comment:
Use repeated `os.path.dirname` instead of `../`

Reply:
I don't see it since the data directory is under root while the ```compare_*.py``` are under root's tests directory. It also works when I run from any location under root.

Reply:
Either works, but the `../` idiom is identical to `dirname`, and there's no reason to mix them. But this is mostly a personal preference; use whichever you would prefer.

Reply:
done.